packages/core/src/orchestrator.ts

```diff
@@ -306,6 +306,7 @@ export function createEnvelopOrchestrator<PluginsContext extends DefaultContext>
           let error: unknown = err;
           for (const errorCb of contextErrorHandlers) {
             errorCb({
+              initialContext,
```

Comment:
I think it might more sense to actually pass `context` here instead of `initialContext`. As you should probably have access to the "last working" context object (before an error got thrown in case you use multiple context extensions).

Reply:
Agreed! I pushed a change.

One more thing, the type support is a bit lack-luster, but it would be quite difficult to type this correctly.

Do you think improving this seems like a bit too much for this Pr?

![Screenshot 2022-04-04 at 12 13 19](https://user-images.githubusercontent.com/2785359/161523632-5ca273f6-0fa0-445f-989d-a507fefd36da.png)

Reply:
his should be addressed as part of https://github.com/dotansimha/envelop/issues/1327